pyanaconda/core/kernel.py

```diff
@@ -110,17 +115,17 @@ def read_string(self, cmdline):
                 val = None
 
             # Some duplicate args create a space separated string
-            if key in CMDLINE_APPEND and self.get(key, None):
+            if key in CMDLINE_APPEND and self._data.get(key, None):
                 if val:
-                    self[key] = self[key] + " " + val
+                    self._data[key] = self._data[key] + " " + val
             # Some arguments can contain spaces so adding them in one string is not that helpful
             elif key in CMDLINE_LIST:
                 if val:
-                    if not self.get(key, None):
-                        self[key] = []
-                    self[key].append(val)
+                    if not self._data.get(key, None):
+                        self._data[key] = []
+                    self._data[key].append(val)
             else:
-                self[key] = val
+                self._data[key] = val
 
     def getbool(self, arg, default=False):
```

Comment:
I wonder whether we shouldn't also get rid of the `getbool` method. We could merge it with `get` and use the logic with the converter from the function `get_option` in `pyanaconda.core.configuration.base`. The converter shouldn't be used if the return value is None (returned as a default).

Reply:
Interesting idea. Just thinking if that won't take part of the expressiveness. However, you probably know what you should expect from the boot option...

Reply:
It's actually easily possibile: There are just 3 uses of `getbool` in Anaconda (geoloc, fips). Most of the boot "nopes" are handled in argument_parsing explicitly with full name, so the magic for "no" prefix is actually mostly useless. Kdump uses `getbool` 3 times, but that's easily fixed too.

I propose removing all _use_ of `getbool` in Anaconda now, and addressing the eradication of external usage in a new work unit. Is that ok?

Reply:
Sounds good to me.